src/components/fx/layout_defaults.js

```diff
@@ -16,6 +16,8 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
         return Lib.coerce(layoutIn, layoutOut, layoutAttributes, attr, dflt);
     }
 
+    coerce('clickmode');
```

Comment:
We're still missing @alexcjohnson smart default suggestion where when a user set `clickmode: 'select'` the default `hovermode` value becomes `'closest'`.

Reply:
Oh, sorry, I misunderstood that. I thought mentioning in `clickmode`'s description that hovermode set to closest is best for most applications was "enough". 

So what we want to have is when `clickmode` includes the `select` flag and nothing is set explicitly for `hovermode`, the default of `hovermode` should be `closest`. Is that right?

Reply:
> So what we want to have is when `clickmode` includes the `select` flag and nothing is set explicitly for `hovermode`, the default of `hovermode` should be `closest`. Is that right? 

Yep! Which should be as simple as modifying the `hovermodeDflt` logic below.

Reply:
Done!